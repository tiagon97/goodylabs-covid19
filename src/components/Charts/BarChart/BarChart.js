import React from 'react';
import { BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer, Cell } from 'recharts';
import Heading from '../../Heading/Heading';

const SimpleBarChart = ({ globalData }) => {
  const data = [
    { name: 'Total confirmed', amount: `${globalData.TotalConfirmed}`, color: '#DAA520' },
    { name: 'Total deaths', amount: `${globalData.TotalDeaths}`, color: 'red' },
    { name: 'Total recovered', amount: `${globalData.TotalRecovered}`, color: '#32CD32' },
  ];

  return (
    <>
      <Heading>Global stats</Heading>
      <ResponsiveContainer width="100%" height={600}>
        <BarChart data={data} margin={{ top: 60, right: 80, left: 80, bottom: 60 }}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip cursor={{ fill: 'none' }} />
          <Bar barSize={250} dataKey="amount" fill="violet">
            {data.map((entry, index) => (
              <Cell key={`cell-${index}`} fill={data[index].color} />
            ))}
          </Bar>
        </BarChart>
      </ResponsiveContainer>
    </>
  );
};

export default SimpleBarChart;
