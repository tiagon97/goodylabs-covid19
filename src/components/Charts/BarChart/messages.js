const messages = {
  heading: 'Global stats',
};

export default messages;
