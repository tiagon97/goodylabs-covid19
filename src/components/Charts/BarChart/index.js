import React from 'react';
import PropTypes from 'prop-types';
import {
  BarChart as RechartsBarChart,
  Bar,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  ResponsiveContainer,
  Cell,
} from 'recharts';

import Heading from '../../Heading';
import formatValue from '../../../utils';
import messages from './messages';
import { Colors } from '../../../constants';

const BarChart = ({ globalData }) => {
  const data = [
    { name: 'Total confirmed', amount: `${globalData.TotalConfirmed}`, color: Colors.yellow },
    { name: 'Total deaths', amount: `${globalData.TotalDeaths}`, color: Colors.red },
    { name: 'Total recovered', amount: `${globalData.TotalRecovered}`, color: Colors.green },
  ];

  return (
    <>
      <Heading>{messages.heading}</Heading>
      <ResponsiveContainer width="100%" height={600}>
        <RechartsBarChart data={data} margin={{ top: 60, right: 80, left: 80, bottom: 60 }}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip cursor={{ fill: 'none' }} formatter={(value) => formatValue`${value}`} />
          <Bar barSize={250} dataKey="amount" fill="royalblue">
            {data.map((_entry, index) => (
              <Cell key={`cell-${_entry}`} fill={data[index].color} />
            ))}
          </Bar>
        </RechartsBarChart>
      </ResponsiveContainer>
    </>
  );
};

BarChart.propTypes = {
  globalData: PropTypes.shape({
    TotalConfirmed: PropTypes.number,
    TotalDeaths: PropTypes.number,
    TotalRecovered: PropTypes.number,
  }),
};

BarChart.defaultProps = {
  globalData: null,
};
export default BarChart;
