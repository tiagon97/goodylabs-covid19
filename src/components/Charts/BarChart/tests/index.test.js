/**
 *
 * Tests for BarChart
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import BarChart from '../index';

const defaultProps = {
  globalData: {
    TotalConfirmed: 100,
    TotalDeaths: 10,
    TotalRecovered: 50,
  },
};

describe('<BarChart />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<BarChart {...defaultProps} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<BarChart {...defaultProps} />);
    expect(firstChild).toMatchSnapshot();
  });
});
