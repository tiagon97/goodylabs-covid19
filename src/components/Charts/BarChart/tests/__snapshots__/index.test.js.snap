// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarChart /> Should render and match the snapshot 1`] = `
<h1
  class="sc-AxjAm cYlDMx"
>
  Global stats
</h1>
`;
